packages/x-date-pickers/src/ClockPicker/ClockPointer.tsx

```diff
@@ -43,59 +69,40 @@ const ClockPointerThumb = styled('div')<{
 
 /**
  * @ignore - internal component.
- * TODO: Remove class
  */
-export class ClockPointer extends React.Component<ClockPointerProps> {
-  static getDerivedStateFromProps = (
-    nextProps: ClockPointerProps,
-    state: ClockPointer['state'],
-  ) => {
-    if (nextProps.type !== state.previousType) {
-      return {
-        toAnimateTransform: true,
-        previousType: nextProps.type,
-      };
-    }
-
-    return {
-      toAnimateTransform: false,
-      previousType: nextProps.type,
-    };
-  };
-
-  state = {
-    toAnimateTransform: false,
-    previousType: undefined,
-  };
+export function ClockPointer(inProps: ClockPointerProps) {
```

Comment:
Refactored to a function component and simplified the additional `state` stored in order to resolve when we `should animate` transition